pipeline/train/train.py

```diff
@@ -257,39 +261,78 @@ def build_datasets(self) -> None:
+        logger.info("trg_script " + repr(trg_script))
+
+        config_input = train_dir / f"configs/opustrainer/{self.model_type.value}.yml"
+        with open(config_input, "rt", encoding="utf-8") as file:
+            config_text = file.read()
+
+        logger.info("Applying OpusTrainer config variables:")
+        for key, value in self.config_variables.items():
+            logger.info(f" {{{key}}}: {value}")
+        config_text = config_text.format(**self.config_variables)
+
+        config_yaml: dict[str, Any] = yaml.safe_load(config_text)
+
+        # Apply the modifiers based on the properties of the language's script.
+        modifier_sections = config_yaml.pop("modifiers")
+        if modifier_sections:
+            logger.info("Applying OpusTrainer modifiers.")
+
+            # The common modifiers are always applied.
+            modifiers = modifier_sections["common"]
 
+            # Bicameral scripts can have their casing augmented.
+            if src_script["bicameral"]:
+                modifiers.extend(modifier_sections["bicameral_src"])
+
+            # Phonemic languages can be misspelled.
+            if is_script_phonemic(src_script["type"]):
+                modifiers.extend(modifier_sections["phonemic_src"])
+
+            # Ensure the "Tags" modifier is last, as it removes the alignments.
+            modifiers.sort(key=lambda modifier: next(iter(modifier)) == "Tags")
+
+            config_yaml["modifiers"] = modifiers
+
+        # Apply the teacher's curriculum.
         if self.model_type == ModelType.teacher:
+            logger.info("Applying OpusTrainer curriculum.")
+            curriculum = config_yaml.pop("curriculum")
             teacher_mode = self.teacher_mode.value
             if teacher_mode == TeacherMode.none.value:
                 raise ValueError("Teacher mode was not properly set, as it was set to none")
+            assert curriculum, "The curriculum must be found."
 
-            config_input = (
-                train_dir
-                / f"configs/opustrainer/{self.model_type.value}.{teacher_mode}.{config_suffix}"
-            )
-        else:
-            config_input = (
-                train_dir / f"configs/opustrainer/{self.model_type.value}.{config_suffix}"
-            )
+            # Apply the curriculum.
+            if teacher_mode not in curriculum:
+                raise ValueError(
+                    f'Could not find the teacher mode "{teacher_mode}" in the curriculum config\n'
+                    + json.dumps(curriculum, indent=2)
+                )
 
-        with open(config_input, "rt", encoding="utf-8") as file:
-            config_text = file.read()
+            config_yaml.update(curriculum[teacher_mode])
 
-        logger.info(f"Applying OpusTrainer config variables: {config_input}")
-        for key, value in self.config_variables.items():
-            logger.info(f" - {key}: {value}")
+        config_text = yaml.safe_dump(config_yaml)
```

Comment:
We need a unit test that tests the generated configs for different cases (compares to the one we expect).

Reply:
Ah, that's a good idea, thanks. I didn't think about the testing picture when writing the code.